src/libclipper/src/selection_policies.cpp

```diff
@@ -60,6 +61,7 @@ std::vector<PredictTask> DefaultOutputSelectionPolicy::select_predict_tasks(
     log_error_formatted(LOGGING_TAG_SELECTION_POLICY,
                         "No candidate models for query with label {}",
                         query.label_);
+    throw NoModelsFoundError();
```

Comment:
This isn't the right place to throw the exception. `select_predict_tasks` can request a query from a model that hasn't been registered with Clipper. In practice, this exception will never get thrown because we never allow an application to be registered that doesn't specify a candidate model. The right place to throw the exception for what you're proposing would be in [`TaskExecutor::schedule_predictions`](https://github.com/ucbrise/clipper/blob/727573caabff5fe8f1cdf4cd8ede0998797e7868/src/libclipper/include/clipper/task_executor.hpp#L339).

But I don't think throwing an exception is the right thing to do here, because that will make the TaskExecutor less forward-compatible when we support multiple models for a query and want to allow for the case where only some models have been registered at this point. See my comment above about checking for `task_futures.empty()` in query_processor.cpp for how I think we should handle this.